src/main/java/reposense/parser/ArgsParser.java

```diff
@@ -164,15 +167,18 @@ public static CliArguments parse(String[] args) throws HelpScreenException, Pars
             Path configFolderPath = results.get(CONFIG_FLAGS[0]);
             Path reportFolderPath = results.get(VIEW_FLAGS[0]);
             Path outputFolderPath = results.get(OUTPUT_FLAGS[0]);
-            Optional<Date> sinceDate = results.get(SINCE_FLAGS[0]);
-            Optional<Date> untilDate = results.get(UNTIL_FLAGS[0]);
+            Optional<Date> cliSinceDate = results.get(SINCE_FLAGS[0]);
+            Optional<Date> cliUntilDate = results.get(UNTIL_FLAGS[0]);
+            Date sinceDate = cliSinceDate.orElse(getSinceDate(cliUntilDate));
+            Date untilDate = cliUntilDate.orElse(getUntilDate());
```

Comment:
Why not do it in `SinceDateArgumentType`? 

If we don't use `Optional` anymore just change `SinceDateArgumentType` to return `Date`?

Reply:
Because if user did not specify `sinceDate`, we will need to rely on `untilDate` to generate the `sinceDate` that is one month before `untilDate`. For that to happen, `cliSinceDate` needs to be initially empty. 

I can't directly get the `sinceDate` from the `convert` method in `SinceDateArgumentType`.

Reply:
> Because if user did not specify `sinceDate`, we will need to rely on `untilDate` to generate the `sinceDate` that is one month before `untilDate`. For that to happen, `cliSinceDate` needs to be initially empty.

Okay, since there is a dependency, we can only handle it after the inputs are parsed.